controllers/velero.go

```diff
@@ -531,3 +530,38 @@ func (r *VeleroReconciler) getVeleroResourceReqs(velero *oadpv1alpha1.Velero) co
 
 	return ResourcesReqs
 }
+
+// For later: Move this code into validator.go when more need for validation arises
+// TODO: if multiple default plugins exist, ensure we validate all of them.
+// Right now its sequential validation
+func (r *VeleroReconciler) ValidateVeleroDeployment(log logr.Logger) (bool, error) {
+	velero := oadpv1alpha1.Velero{}
+	if err := r.Get(r.Context, r.NamespacedName, &velero); err != nil {
+		return false, err
+	}
+
+	secretName := "nosecret"
+	var defaultPlugin oadpv1alpha1.DefaultPlugin
+	for _, defaultPlugin = range velero.Spec.DefaultVeleroPlugins {
+		//Switching only for supported cloud providers
+		switch defaultPlugin {
```

Comment:
As we want to run this validation only for the default cloud provider plugins lets use the following pattern that we have used for cloud provider specific things:
```
for _, plugin = range velero.Spec.DefaultVeleroPlugins {
		if pluginSpecificMap, ok := PluginSpecificFields[plugin]; ok && pluginSpecificMap.IsCloudProvider {
			secretName := pluginSpecificMap.SecretName
		}
	}
```